web-common/src/features/dashboards/time-controls/new-time-controls.ts

```diff
@@ -470,6 +470,19 @@ export function getSmallestUnit(
   return null;
 }
 
+export function getSmallestUnitInDateTime(time: DateTime) {
+  if (time.millisecond) return "millisecond";
+  if (time.second) return "second";
+  if (time.minute) return "minute";
+  if (time.hour) return "hour";
+  if (time.day) return "day";
+  if (time.month) return "month";
+  if (time.quarter) return "quarter";
+  if (time.year) return "year";
+
+  return null;
+}
```

Comment:
It feels a little brittle that this returns a string. Is that the expected type of the consuming code? Or a more specific type?